modules/docs/arrow-docs/docs/docs/optics/syntax/README.md

```diff
@@ -33,6 +29,35 @@ val employee = Employee("John Doe", Company("Kategory", Address("Functional city
 employee.setter().company.address.street.name.modify(String::toUpperCase)
 ```
 
+Arrow can also generate a dsl for a `sealed class` which can be helpful to reduce boilerplate code, or improve readability.
+
+```kotlin:ank:silent
+@optics sealed class NetworkResult
+@optics data class Success(val content: String): NetworkResult()
+@optics sealed class NetworkError : NetworkResult()
+@optics data class HttpError(val message: String): NetworkResult()
+@optics object TimeoutError: NetworkError()
+
+val networkResult: NetworkResult = HttpError("BOOM!")
+```
+
+Let's imagine we have a function `f` of type `(HttpError) -> HttpError` and we want to invoke it on the `NetworkResult`.
+
+```
+val f: (String) -> String = String::toUpperCase
+
+when (networkResult) {
+  is HttpError -> networkResult.copy(f(networkResult.message))
+  else -> networkResult
+}
+```
+
+We can rewrite this code with our generated dsl.
+
+```kotlin:ank
+networkResult.setter().networkError.httpError.message.modify(f)
```

Comment:
why is it `setter().networkError.httpError` and not `setter().httpError` ? How `networkError` is required here if we just want to set the new modified message value when the type is `HttpError` ?

Reply:
That is because of the nested hierarchy of the sealed classes.

We need to compose `Prism<NetworkResult, NetworkError>` with `Prism<NetworkError, HttpError>` to get `Prism<NeworkResult, HttpError>`.

Reply:
Yeah I've noticed now that there's a second level of inheritance there.